### ISSUE TITLE
Compare with Chan-Vese model

### ISSUE THREAD

**Issue by JunMa11** [junyuchen245/Unsuprevised_Seg_via_CNN#1]
Dear @junyuchen245 ,

Thanks for sharing the great work.

How about comparing with the naive Chan-Vese Model in the experiments?

![image](https://user-images.githubusercontent.com/19947331/73325935-5c9e7f80-428b-11ea-8d6f-df46344ec4d5.png)

The idea of this paper is to minimize the energy functonal of CV model by CNN rather than tradition gradient, am I right?

This work you may be interested in:
[Mumford–Shah Loss Functional for Image Segmentation With Deep Learning](https://arxiv.org/pdf/1904.02872v2.pdf)

Best regards,
Jun

**Comment by junyuchen245** [comment]
> Dear @junyuchen245 ,
> 
> Thanks for sharing the great work.
> 
> How about comparing with the naive Chan-Vese Model in the experiments?
> 
> ![image](https://user-images.githubusercontent.com/19947331/73325935-5c9e7f80-428b-11ea-8d6f-df46344ec4d5.png)
> 
> The idea of this paper is to minimize the energy functonal of CV model by CNN rather than tradition gradient, am I right?
> 
> This work you may be interested in:
> [Mumford–Shah Loss Functional for Image Segmentation With Deep Learning](https://arxiv.org/pdf/1904.02872v2.pdf)
> 
> Best regards,
> Jun

Hi @JunMa11,

Thank you for the comments. 

As you can probably tell, this paper was submitted to the MIDL short paper track. The number of pages is limited so that I couldn't include all the details. In general, our method could produce segmentation results that are comparable to the traditional level-set based Chan-Vese model, a figure below shows a quick example: the middle image shows the result generated by our method, and the last image shows the mask obtained with the level-set based Chan-Vese model.

<img width="1051" alt="Screen Shot 2020-01-29 at 12 22 52 AM" src="https://user-images.githubusercontent.com/17306891/73330025-8d4fcc00-422d-11ea-9eac-fbdf7e88be1d.png">

But of course, in reality, you have to tune the hyperparameters in both methods, and different parameter values could lead to different segmentation results. 

One obvious advantage of the CNN based model is that it provides a significant speedup (especially in 3D). Another advantage is that the CNN based model provides the possibility to improve the segmentation performance by fine-tuning the network with a tiny amount of annotated images (as visible in my paper).

[Mumford–Shah Loss Functional for Image Segmentation With Deep Learning](https://arxiv.org/pdf/1904.02872v2.pdf) is certainly a terrific work, and it is similar to what I have done. Eventually, I wish to be able to use Softmax (similar to theirs) in the last layer in the case of multi-class segmentation. However, I somehow failed to converge to a good segmentation result with Softmax. As a result, I was stuck with ReLU, which can only be used for binary class segmentation.

Again, thank you for your comments.

Junyu

**Comment by JunMa11** [comment]
Hi @junyuchen245 ,

Thanks for your valuable reply.

Hope everything goes well with your paper review.

Best,
Jun

**Comment by junyuchen245** [comment]
@JunMa11 Thank you!

Junyu
